associacao-brasileira-de-normas-tecnicas-ufrgs.csl

```diff
@@ -97,35 +111,53 @@
         <name-part name="given" text-case="capitalize-first"/>
         <name-part name="family" text-case="capitalize-first"/>
       </name>
+      <et-al font-style="italic"/>
     </names>
   </macro>
   <macro name="author">
-    <names variable="author">
-      <name name-as-sort-order="all" sort-separator=", " initialize-with=". " delimiter="; " delimiter-precedes-last="always">
-        <name-part name="family" text-case="uppercase"/>
-        <name-part name="given" text-case="capitalize-first"/>
-      </name>
-      <label form="short" prefix=" (" suffix=".)" text-case="uppercase"/>
-      <substitute>
-        <names variable="editor"/>
-        <names variable="translator"/>
-        <text macro="title"/>
-      </substitute>
-    </names>
+    <choose>
+      <if type="motion_picture broadcast song" match="any">
+        <names variable="author">
+          <name>
+            <name-part name="given" text-case="capitalize-first"/>
```

Comment:
no need for name parts elements if you're just going to leave them with capitalize-first.

Reply:
Foi feito porque há mudança na ordem das informações para estes tipos de documentos que tem a entrada do nome de forma direta: nome sobrenome.

Reply:
No, that's a misunderstanding. Using the name-part elements does not affect their order; it just allows you to add things like uppercase, bold, etc. The order of name parts is determined by `<name>` and its attributes. It currently would give you firstname lastname (which I understand is what you want). Changes would be made with `name-as-sort-order`

Reply:
Não conseguimos utilizar o <name-as-sort-order> no <name> para corrigir a ordem dos elementos.

Reply:
We're talking past each other. The `name-parts` elements have simply no relationship to the order of name parts appearing in citations -- try it out: 
If you remove these two lines, the order of name elements stays the same. If you change the order of these two lines, the order of name elements stays the same. 

If you're not able to achieve something you want in the style, please ask, but what you're saying you're doing with the `name-parts`, I guarantee you, they're not actually doing it.

Reply:
Or in other words -- the fact that you have artists displayed as firstname lastname and authors displayed as lastname, firstname is due to the fact that you don't have `name-as-sort-order` here but do have it for authors a couple of lines below.